connexion/decorators/produces.py

```diff
@@ -76,6 +82,81 @@ def get_full_response(data):
                                                                           'url': url})
         return data, status_code, headers
 
+    @staticmethod
+    def process_headers(response, headers):
+        """
+        A convenience function for updating the Response headers with any additional headers
+        generated in the view. If more complex logic should be needed later then it can be handled here.
+
+        :type response: flask.Response
+        :type headers: dict
+        :rtype flask.Response
+        """
+        if headers:
+            for header, value in headers.items():
+                response.headers[header] = value
+        return response
+
+    def get_as_type(self, data_string, schema_type, mimetype):
+        """
+        A function to convert the serialized response body back into its intended types for validation.
+        If the schema is declared as an "object" and the mimetype is a form of JSON, then load as JSON.
+        The string is tested in this order: json, boolean, float, integer - or return as string.
+
+        :type data_string: string The response body as a string
+        :type schema_type: dict The schema type declared in the spec for the response body
+        :rtype dict | bool | float | int | str
+        """
+        if schema_type == "object" and "json" in mimetype:  # json
+            return json.loads(data_string)
+        if data_string[:1] == "[":  # array?
+            try:
+                return ast.literal_eval(data_string)
+            except ValueError:
+                pass
+        if data_string.lower() == "true":  # boolean?
+            return True
+        elif data_string.lower() == "false":
+            return False
+        if "." in data_string:  # float?
+            try:
+                return float(data_string)
+            except ValueError:
+                pass
+        try:  # integer?
+            return int(data_string)
+        except ValueError:
+            pass
+        return data_string  # leave as a string
+
+    def validate_response(self, response, status_code, mimetype):
```

Comment:
It would probably be better to have this a as independent, optional, decorator instead of being inside the `produces` decorator.